docs/query-help-style-guide.md

```diff
@@ -0,0 +1,202 @@
+# Query help style-guide
+
+## Introduction
+
+When you contribute a new query to Semmle/ql for inclusion in the standard queries, you should also write a query help file. This file provides detailed information about the purpose and use of the query, which is available to users in LGTM (for example [here](https://lgtm.com/rules/1506093386171/)) and on the query homepages:
+
+*   [C/C++ queries](https://wiki.semmle.com/pages/viewpage.action?pageId=19334052)
+*   [C# queries](https://wiki.semmle.com/display/CSHARP/C%23+queries)
+*   [COBOL queries](https://wiki.semmle.com/display/COBOL/COBOL+queries)
+*   [Java queries](https://wiki.semmle.com/display/JAVA/Java+queries)
+*   [JavaScript queries](https://wiki.semmle.com/display/JS/JavaScript+queries)
+*   [Python queries](https://wiki.semmle.com/display/PYTHON/Python+queries)
+
+### Location and file name
+
+Query help files must have the same base name as the query they describe and must be located in the same directory.  
+
+### File structure and layout
+
+Query files are written using an XML format called Qhelp, and stored in a file with a `.qhelp` extension. The basic structure is as follows:
+
+```
+<!DOCTYPE qhelp SYSTEM "qhelp.dtd">
+<qhelp>
+    CONTAINS one or more section-level elements
+</qhelp>
+```
+
+The header and single top-level `qhelp` element are both mandatory. 
+
+### Section-level elements
+
+Section-level elements are used to group the information within the query help file. All query help files should include at least the following section elements, in the order specified:
+
+1.  `overview`—a summary of the purpose of the query, including an explanation of what the code does and the how the behavior of the program is affected.
+2.  `recommendation`—information on how to fix the issue highlighted by the query.
+3.  `example`—an example of code showing the problem. Where possible, this section should also include a solution to the issue.
+4.  `references`—relevant references to best practice. 
```

Comment:
Not just references to best practice.  That's of course nice to include, if applicable.  References to authoritative  sources on relevant language semantics should be included, and potentially other relevant references.